lib/Platform/Intl/java/com/facebook/hermes/intl/notes.md

```diff
@@ -0,0 +1,9 @@
+Known Issues
+
+1. "intl402/Collator/ignore-invalid-unicode-ext-values.js" is failing because Array.sort is not stable.
+
+To reproduce, repeat this statement multiple times : 
+
+console.log(testArray.sort(new Intl.Collator().compare));
+
+2. Spec requires the object returned by Intl.Collator.prototype.resolvedOptions() to be in specific order. Because we use JavaHashMaps in Java code and std::unordered_map in the C++ layer, we can't produce deterministic ordering of keys. This fails "intl402/Collator/prototype/order.js"
```

Comment:
Nice catch, I hadn't realized there were were ordering constraints on the resolved options.

I see two potential fixes here.
1. we could make C++ optionsToJs take an extra argument giving the options order, and then it can add them to the JS object in that order.
2. we could LinkedHashMap in Java for resolved options, which preserves insertion order, and in the C++ side, introduce a new OrderedOptions = std::vector<std::pair<std::u16string, Option>> which is used for returning resolved options.

I think 2 is a bit nicer, as it doesn't require maintaining a separate ordering list.

Reply:
I second that #2 is nicer .. In the latest iteration I've fixed the resolvedOptions function of all services to create LinkedHashMap .. the c++ side change is not made though.